vtex/faststore
# Feat: PDP multiple page template

Depends on:

- https://github.com/vtex/faststore/pull/2215
- https://github.com/vtex/vtex-cicd-platform/pull/572
- https://github.com/vtex/faststore/pull/2225

## What's the purpose of this pull request?

This PR aims to find the best PDP template from the CMS, based on the PDP slug and category tree.
[RFC Multiple Page template details](https://docs.google.com/document/d/1qQ573B9iVadSrqjvjSt8ExDu7-uxa3_COXlKvqkTQ8Q/edit)

vendemo account
<img width="1554" alt="Screenshot 2024-02-06 at 17 27 24" src="https://github.com/vtex/faststore/assets/11325562/3bab00e1-09d1-4f07-ba47-c3c6539423ce">

## How it works?

Prioritizing the following order:

1. A PDP template that matches the page slug (e.g. slug = /apple-magic-mouse/p).
2. A PDP template that matches the product subcategory (e.g. /department/category/subcategory).
3. A PDP template that matches the product category (e.g. /department/category).
4. A PDP template that matches the product department (e.g. /department).
5. If no matches are found, use the generic PDP template.

## How to test it?

1. localhost Initial setup:
You can change the tenant from CMS to the account `vendemo`. This store has all the PDPs templates created and published. Also, we changed the stored to `lojavivara`, for the sake of testing with a more complete catalog without harming the devs that are using `storeframework` nowadays.

<img width="924" alt="Screenshot 2024-02-23 at 13 26 42" src="https://github.com/vtex/faststore/assets/11325562/3a3a3654-459d-4761-8b05-50401495bd12">

<img width="796" alt="Screenshot 2024-02-23 at 13 26 02" src="https://github.com/vtex/faststore/assets/11325562/d67c89db-38e6-419d-9fa3-f900ff9dec11">

2. Note that we have different templates created for PDPs with all scenarios mentioned above. So, use the following products to test


|Matching Type|URL|
|-|-|
|Slug - /anel-color-em-prata-925-com-topazio-sky-e-topazios-incolores-an00052637/p | http://localhost:3000/anel-color-em-prata-925-com-topazio-sky-e-topazios-incolores-an00052637/p |
|Subcategory - /vivara/joias/aneis/ |http://localhost:3000/anel-inside-em-prata-925-com-topazios-incolores-an00051429/p |
|Category - /vivara/joias/ |http://localhost:3000/pulseira-riviera-em-prata-925-com-espinelios-pu00016734/p |
|Department - /vivara/ |http://localhost:3000/relogio-tommy-hilfiger-masculino-couro-preto-1710542-to00004729/p |
|General|http://localhost:3000/corrente-para-coleira-life-for-pets-kl00000009/p |

### Starters Deploy Preview

This PR in starter uses the same configs mentioned. `vendemo` for CMS and `lojavivara` for API:
- https://github.com/vtex-sites/starter.store/pull/388

You can test using the same URLs:
|Matching Type|URL|
|-|-|
|Slug - /anel-color-em-prata-925-com-topazio-sky-e-topazios-incolores-an00052637/p | https://sfj-893f2c3--starter.preview.vtex.app/anel-color-em-prata-925-com-topazio-sky-e-topazios-incolores-an00052637/p |
|Subcategory - /vivara/joias/aneis/ |https://sfj-893f2c3--starter.preview.vtex.app/anel-inside-em-prata-925-com-topazios-incolores-an00051429/p |
|Category - /vivara/joias/ |https://sfj-893f2c3--starter.preview.vtex.app/pulseira-riviera-em-prata-925-com-espinelios-pu00016734/p |
|Department - /vivara/ |https://sfj-893f2c3--starter.preview.vtex.app/relogio-tommy-hilfiger-masculino-couro-preto-1710542-to00004729/p |
|General|https://sfj-893f2c3--starter.preview.vtex.app/corrente-para-coleira-life-for-pets-kl00000009/p |


## References

 Doc PR
- https://github.com/vtex/faststore/pull/2236
- [RFC Multiple Page template details](https://docs.google.com/document/d/1qQ573B9iVadSrqjvjSt8ExDu7-uxa3_COXlKvqkTQ8Q/edit)
